contrib/download-frozen-image-v2.sh

```diff
@@ -28,6 +27,9 @@ shift || usage 1 >&2
 [ $# -gt 0 -a "$dir" ] || usage 2 >&2
 mkdir -p "$dir"
 
+# get runtime arch (current for arm64 and amd64 only)
+arch=`uname -m | sed 's/aarch64/arm64/;s/x86_64/amd64/'`
```

Comment:
Any arch detection based on `uname` makes me queasy (I too often build 32bit userspace on a 64bit kernel, for example), so I'd prefer if we used `go env GOARCH` or `dpkg --print-architecture` or something userspace-based for this sort of detection.  One benefit of using `go env GOARCH` would be that it should return the expected values out-of-the-box, and only `arm` will be slightly complicated.

Reply:
Absolutely 👍  "$(go env GOARCH)"